consensus/ibft/src/integration-test/java/org/hyperledger/besu/consensus/ibft/tests/TransitionsTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.consensus.ibft.tests;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.hyperledger.besu.config.BftFork;
+import org.hyperledger.besu.config.JsonUtil;
+import org.hyperledger.besu.consensus.common.bft.BftEventQueue;
+import org.hyperledger.besu.consensus.common.bft.events.NewChainHead;
+import org.hyperledger.besu.consensus.ibft.support.TestContext;
+import org.hyperledger.besu.consensus.ibft.support.TestContextBuilder;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.testutil.TestClock;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+public class TransitionsTest {
+
+  @Test
+  public void transitionsBlockPeriod() throws InterruptedException {
+    final TestClock clock = new TestClock(Instant.EPOCH);
```

Comment:
I prefer this way of handling the clock compared to mocking, it's more explicit - going to steal this for another test!